source/client/index.js

```diff
@@ -56,13 +61,24 @@ module.exports = ({
     })
   })
 
-  const hashIgnoringHistory = withScroll(basedHistory, (_prevLoc, { hash } = {}) => (
-    !hash
-  ))
+  const scrollToHashMiddleware = useScroll((prevRouterProps, { location }) => {
```

Comment:
Uh, this is different... the behaviour used to be "react router, please ignore hashes, but please still go back to the top of the viewport on route change". So now if people want to use a different way of getting scroll to hash working ... this might be a surprise is all.